core/src/worker/mod.rs

```diff
@@ -367,22 +391,23 @@ impl Worker {
             info!("Activity polling is disabled for this worker");
         };
         let la_sink = LAReqSink::new(local_act_mgr.clone());
-        let provider = SlotProvider::new(
+        let slot_provider = SlotProvider::new(
             config.namespace.clone(),
             config.task_queue.clone(),
             wft_semaphore.clone(),
-            external_wft_tx,
+            Arc::new(external_wft_tx),
         );
-        let worker_key = client.workers().register(Box::new(provider));
+        let worker_key = Mutex::new(client.workers().register(Box::new(slot_provider.clone())));
```

Comment:
The provider shouldn't need to be cloned?

Reply:
Currently it is a non-cloneable struct that is moved into this "register" call. But I need to call this "register" call with it again so now I need to have multiple references to this slot provider, so I had to make it cloneable and store it on the worker. I can wrap it in an Arc and update everywhere its used or I can clone it upon use here (or do more advanced stuff w/ references and lifetimes).

Reply:
Ah, yeah, I see since it is getting put into `Self` below, I missed that initially.

I think the alternative here would be to not instantiate the worker key here any longer, since you can create it from within self now that it owns the provider. It probably ends up being the same issue though.

Imo it probably does make sense to wrap the provider in an arc rather than the channel inside it.

Reply:
I was hoping to get away with as little of changes to the existing worker registration/eager code as I could, but I fear we're basically saying I may need to do a bit more to it.

Reply:
It's not that much change, and it's nice to maintain some good stewardship where then structure makes sense rather than a minimum possible change, not like this is in a huge rush or anything.

Reply:
:+1: Done. You're right that it wasn't much change to switch from `Box` to `Arc`: https://github.com/temporalio/sdk-core/pull/721/commits/dfe58a56751267b8dab9b4fd28abea4622e26b29

Reply:
Switched from Box to Arc, you were correct, not a big change (https://github.com/temporalio/sdk-core/commit/417b70959bc872aa8c5f451002b1057d5a32026a).